core/sql/executor/ex_sort.h

```diff
@@ -356,4 +363,119 @@ class ExSortFromTopPrivateState : public ex_tcb_private_state
   ~ExSortFromTopPrivateState();       // destructor
 };
 
+class ExSortBufferPool : public NABasicObject
```

Comment:
A more object-oriented way to do this is to create a class hierarchy, with ExSortBufferPool as the abstract base class, and two specializations, one for simple buffers and the other for SQL buffers. Then all your methods are virtual and you get rid of all the "if (sqlBufferPool_)" checks. The code will be slightly more efficient too though probably not significantly so.